clang/lib/Sema/SemaDecl.cpp

```diff
@@ -12233,7 +12233,6 @@ void Sema::CheckMain(FunctionDecl *FD, const DeclSpec &DS) {
        FD->getDeclContext()->getRedeclContext()->isTranslationUnit())) {
     Diag(FD->getLocation(), diag::ext_main_invalid_linkage_specification)
         << FD->getLanguageLinkage();
-    FD->setInvalidDecl();
     return;
```

Comment:
Should we proceed with other checks after this one?

Reply:
@alexfh Thanks for the review. I've removed it to proceed with other checks...